app/controllers/SchemeDetailsController.scala

```diff
@@ -97,19 +105,29 @@ class SchemeDetailsController @Inject()(appConfig: FrontendAppConfig,
                 Ok(schemeDetails(appConfig,
                   schemeName,
                   openedDate(srn.id, list, isSchemeOpen),
-                  administratorsVariations(request.psaId.id, scheme, schemeStatus),
+                  administratorsVariations(request.psaId.id, result._1, schemeStatus),
                   srn.id,
-                  isSchemeOpen
+                  isSchemeOpen,
+                  isLocked
                 ))
               }
             }
           }
         } else {
           Future.successful(NotFound(errorHandler.notFoundTemplate))
         }
-      }
     }
 
+
+  private def withSchemeAndLock(srn: SchemeReferenceNumber)(implicit request: AuthenticatedRequest[AnyContent]) = {
```

Comment:
Just a suggestion - if we define withSchemeAndLock as below we can avoid using tuple as return type and thus make the code a bit more readable:
private def withSchemeAndLock(srn: SchemeReferenceNumber)(xx:(UserAnswers,Option[Lock]) => Future[Result])(implicit request: AuthenticatedRequest[AnyContent])

I.e. the calling code would look like :-1: 
  private def onPageLoadVariations(srn: SchemeReferenceNumber)(implicit request: AuthenticatedRequest[AnyContent]): Future[Result] =
    withSchemeAndLock(srn){ (userAnswers,lock) =>

 .....

    }

Reply:
its not working with above however fixed with below approach
withSchemeAndLock(srn).flatMap{  case (userAnswers,lock) =>